internal/xds/translator/httpfilters.go

```diff
@@ -344,5 +345,26 @@ func patchResources(tCtx *types.ResourceVersionTable, routes []*ir.HTTPRoute) er
 			return err
 		}
 	}
+
+	// create the envoy client TLS cert secret.
+	// it's only used for establishing TLS connections with the rate limit server for now.
+	return createRateLimitClientTLSCertSecret(tCtx, routes)
+}
+
+func createRateLimitClientTLSCertSecret(tCtx *types.ResourceVersionTable, routes []*ir.HTTPRoute) error {
+	for _, route := range routes {
```

Comment:
if cert was a pointer, we could have skipped a lot of this creation, and only done it for gateway ns mode

Reply:
I had the same thought while writing this. Another approach could be we make the envoy secret global so we only need to create it once.